tests/deploy/e2e/test_autostarted.py

```diff
@@ -315,3 +360,720 @@ async def test_autostart_clear_agent_venv_on_delete(
+        assert should_time_out, f"This wasn't supposed to time out with a deployment sleep set to {time_to_sleep}!"
+    finally:
+        # Retrieve the current processes, we should have more processes than `start_children`
+        state_after_deployment = construct_scheduler_children(current_pid)
+
+    expected_additional_children_after_deployment = 3
+    assert len(state_after_deployment.children) == expected_additional_children_after_deployment, (
+        "These processes should be present: The fork server and the actual agent! " f"Actual state: {state_after_deployment}"
+    )
+    for child in state_after_deployment.children:
+        assert child.is_running()
+
+    # The number of resources for this version should match
+    result = await client.list_versions(tid=environment)
+    assert result.code == 200
+    assert len(result.result["versions"]) == 1
+    assert result.result["versions"][0]["total"] == 1
+
+    # Let's check the agent table and check that agent1 is present and not paused
+    await assert_is_paused(client, environment, {const.AGENT_SCHEDULER_ID: False, "agent1": False})
+
+    # Now let's halt the environment
+    result = await client.halt_environment(tid=environment)
+    assert result.code == 200
+
+    snippetcompiler.setup_for_snippet(
+        f"""
+    import minimalwaitingmodule
+
+    a = minimalwaitingmodule::Sleep(name="test_sleep", agent="agent1", time_to_sleep={time_to_sleep})
+    """,
+        autostd=True,
+    )
+    version, res, status = await snippetcompiler.do_export_and_deploy(include_status=True)
+    result = await client.release_version(environment, version, push=False)
+    assert result.code == 200
+
+    # Let's recheck the agent table and check that the scheduler and agent1 are present and paused
+    await assert_is_paused(client, environment, {const.AGENT_SCHEDULER_ID: True, "agent1": True})
+
+    # Let's wait for the executor to die (
+    await retry_limited(
+        wait_for_terminated_status,
+        timeout=const.EXECUTOR_GRACE_HARD + 2,
+        current_children=state_after_deployment.children,
+        expected_terminated_process=3,
+    )
+
+    # Let's recheck the number of processes after pausing the environment
+    halted_state = construct_scheduler_children(current_pid)
+
+    assert (
+        len(halted_state.children) == 0
+    ), "The Scheduler and the fork server and the agent created by the scheduler should have been killed!"
+    for child in halted_state.children:
+        assert child.is_running()
+
+    await client.resume_environment(environment)
+
+    # Let's check the agent table and check that the scheduler and agent1 are present and not paused
+    await assert_is_paused(client, environment, {const.AGENT_SCHEDULER_ID: False, "agent1": False})
+
+    if should_time_out:
+        # Wait for at least one resource to be in deploying
+        await retry_limited(are_resources_being_deployed, timeout=5)
+        await asyncio.sleep(0.5)
```

Comment:
I did this because I had this assertion error (forked executor had the same name of the fork server for a small amount of time):
```
            match child.name():
                case "inmanta: multiprocessing fork server":
>                   assert not fork_server
E                   AssertionError: assert not psutil.Process(pid=81210, name='inmanta: multiprocessing fork server', status='sleeping', started='14:55:36')

/home/hugo/Documents/Inmanta/copies/inmanta-core/tests/deploy/e2e/test_autostarted.py:439: AssertionError
-> assert not fork_server
(Pdb) children
[psutil.Process(pid=81209, name='python', status='sleeping', started='14:55:36'), psutil.Process(pid=81210, name='inmanta: multiprocessing fork server', status='sleeping', started='14:55:36'), psutil.Process(pid=81213, name='inmanta: executor process 78f726cec1879275a4cc2fa2c103afb5 - connected', status='sleeping', started='14:55:36')]
(Pdb) fork_server
psutil.Process(pid=81210, name='inmanta: multiprocessing fork server', status='sleeping', started='14:55:36')
(Pdb) child
psutil.Process(pid=81213, name='inmanta: executor process 78f726cec1879275a4cc2fa2c103afb5 - connected', status='sleeping', started='14:55:36')
(Pdb) 
```
Is there a better (or cleaner) solution to fix this?